File: Bubble_Sort/bubblesort.c
#include <stdbool.h>
#include <stdio.h>

void swap(int* xp, int* yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void bubbleSort(int arr[], int n)
{
	bool swapped;
	for (int i = 0; i < n - 1; i++) {
		swapped = false;
		for (int j = 0; j < n - i -1; j++) {
			if (arr[j] > arr[j + 1]) {
				swap(&arr[j], &arr[j + 1]);
				swapped = true;
			}
		}

		if (swapped == false)
			break;
	}
}

void printArray(int arr[], int size)
{
	for (int i = 0; i < size; i++) {
		printf("%d\n", arr[i]);
	}
}

int main() {
	int arr[] = {64, 56, 87, 99, 69, 11};
	int n = sizeof(arr) / sizeof(arr[0]);
	bubbleSort(arr, n);
	printf("Sorted Array\n");
	printArray(arr, n);
	return 0;
}

File: Bubble_Sort/go/bubblesort.go
package main

import (
	"fmt"
)

func bubbleSort(array []int) []int {
	array_size := len(array)

	for i := 0; i < array_size; i++ {
		swapped := false

		for j := 0; j < array_size-i-1; j++ {
			if array[j] > array[j+1] {
				// Swap the elements
				array[j], array[j+1] = array[j+1], array[j]
				swapped = true
			}
		}

		if !swapped {
			break
		}
	}

	return array
}

func main() {
	arr := []int{64, 34, 25, 12, 22, 11, 90}
	sortedArr := bubbleSort(arr)
	fmt.Println("Sorted array:", sortedArr)
}
